sdk/ai/azure-ai-client/azure/ai/client/operations/_operations.py

```diff
@@ -2231,72 +2231,78 @@ def create_thread(
         :rtype: ~azure.ai.client.models.AgentThread
         :raises ~azure.core.exceptions.HttpResponseError:
         """
-        error_map: MutableMapping = {
-            401: ClientAuthenticationError,
-            404: ResourceNotFoundError,
-            409: ResourceExistsError,
-            304: ResourceNotModifiedError,
-        }
-        error_map.update(kwargs.pop("error_map", {}) or {})
 
-        _headers = case_insensitive_dict(kwargs.pop("headers", {}) or {})
-        _params = kwargs.pop("params", {}) or {}
+        with start_create_thread_span(self._config.project_name, messages=messages) as span:
```

Comment:
this is where monkey-patching comes really handy - helps to avoid customizing autogenerated code